Add coupon generator and option to orders to add a coupon to each order

louwie17 opened the pull request:
Fixes #68 

### Description

Added a Coupon generator with a min and max option for the `int_random` function.

I also added a `coupon` option to the order generator CLI command, which if set, it will auto generate a coupon for each order.

### Testing instructions

- Run `wp wc generate coupons 10 --min=10 --max=40`
- Go to **Marketing > Coupons** and see if there are 10 new coupons (no duplicates)
- Run `wp wc generate orders 10 --status=completed --coupons=true`
- Run `wp action-scheduler run`
- Go to **Analytics > Coupons** and see if there are 10 coupons listed with an order each

### Changelog entry

> Add: coupon generator and a new option for orders to allow for coupon generation.

louwie17 committed:
Add coupon generator and option to orders to add a coupon to each order

louwie17 committed:
Make sure we do not create duplicate coupons

rrennick commented:
@louwie17 Thanks for the PR. Could you add a change log entry to your PR description? There is also a merge conflict that needs to be addressed.

louwie17 committed:
Fix linting

louwie17 commented:
> @louwie17 Thanks for the PR. Could you add a change log entry to your PR description? There is also a merge conflict that needs to be addressed.

Hi @rrennick, I added the change log entry and rebased the branch.

rrennick commented on includes/Generator/Order.php:
```diff
@@ -61,11 +61,17 @@ public static function generate( $save = true, $assoc_args = array() ) {
 
 		$order->set_date_created( $date );
 
+		$include_coupon = ! empty( $assoc_args['status'] );
```
Should this be `$assoc_args['coupons']`?

rrennick commented on includes/Generator/Coupon.php:
```diff
@@ -0,0 +1,42 @@
+	 * @return \WC_Customer Customer object with data populated.
+	 */
+	public static function generate( $save = true, $min = 5, $max = 100 ) {
+		self::init_faker();
```
t doesn't look like the faker is used in this function.

rrennick reviewed (changes requested):
@louwie17 This tested great for me. I only had the two minor issues that need to be looked at. Once those are fixed I'll be able to merge this.

louwie17 committed:
Fix orders CLI param and remove unused faker call

louwie17 commented:
> @louwie17 This tested great for me. I only had the two minor issues that need to be looked at. Once those are fixed I'll be able to merge this.

Thanks for the review @rrennick, great catch on both of those, I have pushed up those changes in this commit: https://github.com/woocommerce/wc-smooth-generator/pull/71/commits/1d497d44267f79b714d516b1c64e87906eec5389

rrennick reviewed (approved):
@louwie17 Thanks for the help.
